webpack/environment.js

```diff
@@ -1,6 +1,30 @@
 module.exports = {
     I18N_HASH: 'generated_hash',
     SERVER_API_URL: '',
-    __VERSION__: process.env.hasOwnProperty('APP_VERSION') ? process.env.APP_VERSION : 'DEV',
+    __VERSION__: process.env.APP_VERSION || inferVersion(),
     __DEBUG_INFO_ENABLED__: false,
 };
+
+/*
+ * Needed for client compilations with docker-compose, where the 'APP_VERSION' property isn't injected by gradle.
+ *
+ * Returns the inferred APP_VERSION from 'build.gradle', or 'DEV' if this couldn't be retrieved
+ */
+function inferVersion() {
+    let version = 'DEV';
+
+    try {
+        const fs = require('fs');
+        const buildGradleFile = 'build.gradle';
+
+        let data = fs.readFileSync(buildGradleFile, 'UTF-8');
+
+        version = data.match(/\nversion\s=\s"(.*)"/);
+
+        version = version[1] ?? 'DEV';
+    } catch (error) {
+        console.log("Error while retrieving 'APP_VERSION' property");
+    }
+
+    return version;
```

Comment:
That's minor, but I don't like the fact that `DEV` is repeated twice in the code.:
```suggestion
    const dev = 'DEV';
    try {
        const fs = require('fs');
        const buildGradleFile = 'build.gradle';

        const data = fs.readFileSync(buildGradleFile, 'UTF-8');

        const version = data.match(/\nversion\s=\s"(.*)"/);

        return version[1] ?? dev;
    } catch (error) {
        console.log("Error while retrieving 'APP_VERSION' property");
    }
    return dev;
```